src/main/java/org/prebid/server/auction/requestfactory/Ortb2RequestFactory.java

```diff
@@ -0,0 +1,287 @@
+     */
+    private String parentAccountIdFromExtPublisher(ExtPublisher extPublisher) {
+        final ExtPublisherPrebid extPublisherPrebid = extPublisher != null ? extPublisher.getPrebid() : null;
+        return extPublisherPrebid != null ? StringUtils.stripToNull(extPublisherPrebid.getParentAccount()) : null;
+    }
+
+    private Future<Account> responseForEmptyAccount(RoutingContext routingContext) {
+        EMPTY_ACCOUNT_LOGGER.warn(accountErrorMessage("Account not specified", routingContext), 100);
+        return responseForUnknownAccount(StringUtils.EMPTY);
+    }
+
+    private static String accountErrorMessage(String message, RoutingContext routingContext) {
+        final HttpServerRequest request = routingContext.request();
+        return String.format(
+                "%s, Url: %s and Referer: %s",
+                message,
+                request.absoluteURI(),
+                request.headers().get(HttpUtil.REFERER_HEADER));
+    }
+
+    private Future<Account> accountFallback(Throwable exception,
+                                            String accountId,
+                                            RoutingContext routingContext) {
+
+        if (exception instanceof PreBidException) {
+            UNKNOWN_ACCOUNT_LOGGER.warn(accountErrorMessage(exception.getMessage(), routingContext), 100);
+        } else {
+            logger.warn("Error occurred while fetching account: {0}", exception.getMessage());
+            logger.debug("Error occurred while fetching account", exception);
+        }
+
+        // hide all errors occurred while fetching account
+        return responseForUnknownAccount(accountId);
+    }
+
+    private Future<Account> responseForUnknownAccount(String accountId) {
+        return enforceValidAccount
+                ? Future.failedFuture(new UnauthorizedAccountException(
+                String.format("Unauthorized account id: %s", accountId), accountId))
+                : Future.succeededFuture(Account.empty(accountId));
+    }
+
+    private Future<Account> ensureAccountActive(Account account) {
+        final String accountId = account.getId();
+
+        return account.getStatus() == AccountStatus.inactive
+                ? Future.failedFuture(new UnauthorizedAccountException(
+                String.format("Account %s is inactive", accountId), accountId))
+                : Future.succeededFuture(account);
+    }
+
+    /**
+     * Performs thorough validation of fully constructed {@link BidRequest} that is going to be used to hold an auction.
+     */
+    public BidRequest validateRequest(BidRequest bidRequest) {
+        final ValidationResult validationResult = requestValidator.validate(bidRequest);
+        if (validationResult.hasErrors()) {
+            throw new InvalidRequestException(validationResult.getErrors());
+        }
+        return bidRequest;
+    }
+
+    public BidRequest enrichBidRequestWithAccountAndPrivacyData(BidRequest bidRequest,
```

Comment:
Let's move public method to the top of the class so that all public methods are followed by all private methods.